nucypher/blockchain/eth/actors.py

```diff
@@ -1632,7 +1633,7 @@ def last_committed_period(self) -> int:
 
     @only_me
     @save_receipt  # saves txhash instead of receipt if `fire_and_forget` is True
-    def commit_to_next_period(self, fire_and_forget: bool = True) -> TxReceipt:
+    def commit_to_next_period(self, fire_and_forget: bool = True) -> Union[TxReceipt, HexBytes]:
```

Comment:
I remember bumping into this pattern before, and currently in our code the propagation of these types is... quite disordered. Like, `staking_agent.commit_to_next_period()` claims to return `Union[TxReceipt, HexBytes]`, but the return value in the function body is explicitly marked as `TxReceipt`, which is in turn obtained from a function that is marked as returning `dict`, in which the return value is obtained from a function returning `Union[TxReceipt, HexBytes]`. So... it probably deserves a PR by itself.

Reply:
When the `fire_and_forget` feature was added, a receipt of transaction hash can be returned.  Not surprised if there is *hint* inaccuracy.

Reply:
> Like, `staking_agent.commit_to_next_period()` claims to return `Union[TxReceipt, HexBytes]`, but the return value in the function body is explicitly marked as `TxReceipt`, which is in turn obtained from a function that is marked as returning `dict`, in which the return value is obtained from a function returning `Union[TxReceipt, HexBytes]`

This was addressed in this PR, at least partially (see 35af6a3a07ee73eaa530198df1a05342a1de438c). I just included some other type hints there as well.